build-system/encrypt-secret.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+START_BOLD='\033[1m'
+END_BOLD='\033[0m'
+PROJECT_NAME=`gcloud info | grep -oP '(?<=project: \[)[-\w]+'`
```

Comment:
Does this script require gcloud setup before running? If so, can we pass in a project name instead?

Reply:
Yes, as do all the project and keyring set-up commands. In theory we could try to automatically run `gcloud init`, authenticate, and select the project ID. In practice, I think that's more than this script should be responsible for. The developer will already need to have used `gcloud init` to set up the project and keyring, and would need to do so any time they wanted to run `gcloud app deploy` anyway. At most I could see accepting the project name and failing on an error if that doesn't match the currently active project configuration

Reply:
I was thinking about the use case where a project and keyring was already set up but you wanted to add an env variable. I just saw `gcloud kms encrypt` in the script, which I'm not sure needs `gcloud init`, but I'm ok with this as is.

Reply:
I think it does, because you have to have access permissions to the key/keyring (which, afaik, is only possible by authenticating within `gcloud init`)